cradlepoint/pages/api/getTestPlan.js

```diff
@@ -4,10 +4,65 @@ import connectToDb from "../../util/mongodb";
   Gets the requested test plan from the database
 */
 export default async (req, res) => {
-  const query = { '_id': ObjectId(req.query.testPlanId) };
 
   const client = await connectToDb();
-  const cursor = client.collection("testPlan").find(query);
+  const cursor = client.collection("testPlan").aggregate([
+    // Find the test Plan
+    { $match: { "_id": ObjectId(req.query._id) } },
```

Comment:
Same as other comment.

Reply:
as above, I left this as Id since we are pulling a test plan by its Id and not by another element's id